From 8c06356cc27662a665be2c5b5defdd3a59b3b1fe Mon Sep 17 00:00:00 2001
From: Lauri Nurmi <lanurmi@iki.fi>
Date: Fri, 27 Feb 2015 23:45:50 +0200
Subject: [PATCH] Use proper Apple-like menu labels for
 wxID_{ABOUT,PREFERENCES,EXIT} and prevent overriding them with generic stock
 labels or custom values.

---
 src/osx/menuitem_osx.cpp | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/src/osx/menuitem_osx.cpp b/src/osx/menuitem_osx.cpp
index 34a9996..7b4cff8 100644
--- a/src/osx/menuitem_osx.cpp
+++ b/src/osx/menuitem_osx.cpp
@@ -160,6 +160,16 @@ void wxMenuItem::Check(bool bDoCheck)
 
 void wxMenuItem::SetItemLabel(const wxString& text)
 {
+    // sensible, apple-like labels for these stock ids are set
+    // by CreateAppleMenu() -- prevent overriding those by
+    // generic stock labels or custom values.
+    if (GetId() == wxID_ABOUT ||
+        GetId() == wxID_PREFERENCES ||
+        GetId() == wxID_EXIT)
+    {
+        return;
+    }
+
     // don't do anything if label didn't change
     if ( m_text == text )
         return;
-- 
1.8.3.msysgit.0

